sqllogictest/src/runner.rs

```diff
@@ -17,7 +17,7 @@ use crate::parser::*;
 #[async_trait]
 pub trait AsyncDB: Send {
     /// The error type of SQL execution.
-    type Error: std::error::Error + Send + 'static;
+    type Error: std::error::Error + Send + Sync + 'static;
```

Comment:
Why to change this?

Reply:
Because `ParallelTestError` now contains `TestError` instead of string. We need this `Sync` constraint otherwise `ParallelTestError` will become `!Sync` and break the compile of risingwave simulation.

Reply:
I feel that most error types are both `Send` and `Sync`. So it is reasonable to do this change?